### ISSUE TITLE
Aligned formatting

### ISSUE THREAD

**Issue by lionel-** [posit-dev/air#114]
Related to #113. A special comment like `fmt: align-equal` would apply to the next expression (which should be a function call) and instruct Air to align arguments on `=`:

```r
list(
    foo = 1,
      x = 2,
          3
)
```

This would help readability of calls that are used for certain types of data entry.

**Comment by DavisVaughan** [comment]
See also aligned assignments https://github.com/posit-dev/air/issues/253

**Comment by januz** [comment]
This is what is keeping me from using Air at the moment (thought I'm eager to use it) because I like to align assignments for readability. If you're getting to work on this, it would be fantastic if this could be a global setting that leaves aligned assignments untouched (or potentially even formats them to be aligned if they are not; though I think that will likely be harder or impossible to implement globally). Thanks!

**Comment by DavisVaughan** [comment]
@januz it seems highly unlikely that we would ever do this globally, sorry. This is more for one off function calls that you want to be aligned (often due to it being some kind of data entry)

**Comment by januz** [comment]
>  when you have repeated information spread across multiple lines, it can help

Yes, for things like defining `tribble()`s, it's absolutely essential. I would argue that it helps with readability throughout though which you seem to acknowledge in your [examples](https://posit-dev.github.io/air/formatter.html#skip-comments) for where one would use `fmt: skip`. The list example is how I write throughout and it would be impractical to use `fmt: skip` everywhere which is why this and the other issue about alignment exist.

> I'd argue that's just part of "Personal Style versus Consistency" from the book, and is exactly what you should do

Hehe, yeah, I guess that's the easiest solution. You are certainly right in that it is harder to get a team aligned on a more complicated (but arguably better / more readable) coding style like this one (I have tried). But it's a bit of a chicken and egg problem as in if the formatter does not support alternative styles, then obviously the easiest style to enforce is the one that the formatter supports.

**Comment by januz** [comment]
@lionel- Yes, I understand your rationale! I think that I'll end up accepting this and giving up alignment to be able to use Air with my team. Standardization and automation of style are the most important thing and Air does a great job with it. Thanks!

**Comment by DavisVaughan** [comment]
We really appreciate your open mindedness about this 😄

**Comment by lionel-** [comment]
@teunbrand We are unlikely to support this sort of alignment I think. The reason we might consider it for calls is for the data entry cases.

**Comment by lionel-** [comment]
@hadley shared an interesting project with relevant ideas: https://github.com/j-brooke/FracturedJson/wiki

**Comment by lorenzwalthert** [comment]
I'd welcome it if alignment detection worked as similar as possible in air as in {styler}. Otherwise, these two formatters might destroy each others' alignment. I welcome you to have a look at what I built back in around 2019 and see if there is anything that makes sense for air as well. There are docs at https://styler.r-lib.org/articles/detect-alignment.html, tests, source code (maybe this could speed up implementing such a feature with LLMs substantially). Alignment detection is on by default, I not even sure you can turn it off, had to go back and check...
